File: app/src/main/java/cenergy/central/com/pwb_store/manager/service/HDLService.java
package cenergy.central.com.pwb_store.manager.service;

import cenergy.central.com.pwb_store.model.AvaliableStoreDao;
import cenergy.central.com.pwb_store.model.ShippingDao;
import cenergy.central.com.pwb_store.model.request.HDLRequest;
import cenergy.central.com.pwb_store.model.request.ShippingRequest;
import cenergy.central.com.pwb_store.model.response.HDLResponse;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.POST;
import retrofit2.http.Query;

/**
 * Created by napabhat on 9/22/2017 AD.
 */

public interface HDLService {

    @POST("/HDLAPI/HomeDelivery/CheckTimeSlot")
    Call<HDLResponse> checkTimeSlot(
            @Header("Authorization") String appSecret,
            @Header("Content-Type") String type,
            @Body HDLRequest hdlRequest
    );

    @POST("v1/logistics/shipment/list-shipping-slot")
    Call<ShippingDao> checkShippingTime(
            @Body ShippingRequest shippingRequest
            );

    @GET("v1/products/list-stock")
    Call<AvaliableStoreDao> getStore(
            @Query("skus") String sku
    );
}

File: app/src/main/java/cenergy/central/com/pwb_store/manager/service/HDLService.kt
package cenergy.central.com.pwb_store.manager.service

import cenergy.central.com.pwb_store.model.ShippingSlot
import cenergy.central.com.pwb_store.model.body.BookingSlotBody
import cenergy.central.com.pwb_store.model.response.HDLMemberResponse
import retrofit2.Call
import retrofit2.http.*

interface HDLService {
    @GET("/v1/customers/{number}")
    fun getHDLMembers(@Path("number") number: String,
                      @Header("cxfTrace") cxfTrace: Boolean): Call<HDLMemberResponse>

    // mdc
    @PUT("/rest/V1/guest-carts/{cartId}/shipping-slot-hdl/book")
    fun createBooking(@Header("client") client: String,
                      @Header("client-type") clientType: String,
                      @Header("retailer-id") retailerId: String,
                      @Path("cartId") cartId: String,
                      @Body bookingSlotBody: BookingSlotBody): Call<ShippingSlot>
}
